kubernetes/istio_details_service.go

```diff
@@ -2,8 +2,10 @@ package kubernetes
 
 import (
 	"fmt"
-	"github.com/kiali/kiali/config"
 	"sync"
+
+	"github.com/kiali/kiali/config"
+	"github.com/kiali/kiali/util/hostname"
```

Comment:
Not sure if the util package is the best one.
I guess this logic makes more sense to have it in the kubernetes package as it is related to istio/kubernetes objects.
It doesn't mean it can't be exported, on the contrary, but as this is to match a specific field of Istio object, I won't search for it under the util one, that's confusing IMO.

Reply:
You are totally right, this shouldn't be used outside kubernetes layer.